frontend/src/components/registerview/RegisterView.jsx

```diff
@@ -1,50 +1,87 @@
 import React, { Component } from 'react';
-import { Form, Button } from 'react-bootstrap'
+import { Form, Button, InputGroup, FormControl } from 'react-bootstrap'
 import AuthService from "../../services/AuthService";
-import {login} from "../../redux/authActions";
+import { login } from "../../redux/authActions";
 import store from "../../redux-store";
 
 class RegisterView extends Component {
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = {
       username: '',
-      password: ''
+      password: '',
+      confirm_password: '',
+      password_type: 'password'
```

Comment:
- [ ] Comparing password_type as string looks like an antipattern to me. If we have only two statuses, I would suggest using boolean i.e. state.passwordVisible, and check if it is true. If we have more than 2 statuses, I would create an **enum** to store them. One way to use enum in JS is to simply [create a JS object and freeze it](https://stackoverflow.com/questions/287903/what-is-the-preferred-syntax-for-defining-enums-in-javascript)